### PR TITLE
refactor(pfd): move to MSFS avionics framework

### PR DESCRIPTION
**Pull request by Saschl**
Fixes #[issue_no]

## Summary of Changes

Re-Implementation of the current React PFD with the MSFS avionics framework for improved performance.

TODO:
 

- [x]  port all existing functionality
- [x] port fpv + RA changes from newer Pull Requests
- [x] cleanup
- [x] proper integration into build pipeline 
   - [x] How to integrate the framework (clone, include, submodule??)
   - [x] unite msfs types
   - [x] unite package.json and node_modules if possible
   - [x] fix eslint config
- [x] pick up svg changes from master
- [x] adapt speed trend changes once merged
- [x] remove duplicated code
- [x] unify component manipulation if possible (subscription, eventbus, setAttribute, classlist...)
- [x] check possibility to use `withPrecision` on arinc values, Partly done, to be improved later on.

## Additional context

Should enable an overall smoother experience with less stutters

Discord username (if different from GitHub):

## Testing instructions

1. A bit difficult to say as the behaviour of the PFD should be the same as it is with the development version currently.
2. Make sure there is no missing symbols, texts (especially the FMA).
3. You might not always see higher FPS, but the experience should be smoother in comparison to the development version.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by beheh** [comment]
@Saschl Heads up: #6736 is now in master and brought a few changes to the PFD - namely ARINC words for the RA and filtered radio altitudes.

**Comment by Saschl** [comment]
Thanks, already started porting :) Will let you know once I face challenges.

File: src/PFDV2/pfd/shared/PFDSimvarPublisher.tsx [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    potentiometer_captain= 'LIGHT POTENTIOMETER:88',
+    potentiometer_fo= 'LIGHT POTENTIOMETER:90',
+
+    pitch = 'L:A32NX_ADIRS_IR_1_PITCH',
+    roll = 'L:A32NX_ADIRS_IR_1_ROLL',
+    heading = 'L:A32NX_ADIRS_IR_1_HEADING',
+    altitude = 'L:A32NX_ADIRS_ADR_1_ALTITUDE',
+    speed = 'L:A32NX_ADIRS_ADR_1_COMPUTED_AIRSPEED',
+    alpha_prot = 'L:A32NX_SPEEDS_ALPHA_PROTECTION',
+    onGround = 'SIM ON GROUND',
```
Comment (tracernz) [inline src/PFDV2/pfd/shared/PFDSimvarPublisher.tsx:114]: We have LGCIU discrete vars for this now... https://github.com/flybywiresim/a32nx/blob/master/docs/a320-simvars.md#landing-gear-ata-32

File: src/PFDV2/pfd/shared/PFDSimvarPublisher.tsx [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    fdPitch = 'L:A32NX_FLIGHT_DIRECTOR_PITCH',
+    v1 = 'L:AIRLINER_V1_SPEED',
+    vr = 'L:AIRLINER_VR_SPEED',
+    fwcFlightPhase = 'L:A32NX_FWC_FLIGHT_PHASE',
+    fmgcFlightPhase = 'L:A32NX_FMGC_FLIGHT_PHASE',
+    hasLoc = 'L:A32NX_RADIO_RECEIVER_LOC_IS_VALID',
+    hasDme = 'NAV HAS DME:3',
+    navIdent = 'NAV IDENT:3',
+    navFreq = 'NAV FREQUENCY:3',
+    dme = 'NAV DME:3',
```
Comment (tracernz) [inline src/PFDV2/pfd/shared/PFDSimvarPublisher.tsx:151]: Note these are changed on master...

File: scripts/setup.sh [hunk 3]
```diff
@@ -7,12 +7,14 @@ cd msfs-avionics-mirror/src/sdk
...
 npm pack
 
-cd /external
-rm -f package-lock.json
-rm -rf node_modules
-npm install --no-optional
+#cd /external
+#rm -f package-lock.json
+#rm -rf node_modules
+#npm install --no-optional
```
Comment (beheh) [inline scripts/setup.sh]: These look like debug changes and should probably not be in this PR.

**Review by beheh — CHANGES_REQUESTED** [review]

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@ jobs:
       - name: Checkout source
         uses: actions/checkout@v2
       - name: install
-        run: npm install --no-optional
+        run: ./scripts/dev-env/run.sh ./scripts/setup.sh
```
[the remark this answers is not in the document]
Comment (Saschl) [reply .github/workflows/pr.yml:18]: As we build the WT framework as part of the `setup.sh` process, the current state would fail because it cannot find the framework when running `npm install`

File: .gitignore [hunk 5]
```diff
@@ -62,3 +63,8 @@ src/instruments/buildSrc/custom/*
 /src/mcdu-server/client/build
 !/flybywire-aircraft-a320-neo/CORS SERVER/*
 /flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/TEXTURE/A320NEO_COCKPIT_DECALSTEXT_ALBD.TIF.dds
+/**/.rollup.cache
+tsconfig.tsbuildinfo
+/**/package-lock.json
+msfs-avionics-mirror/src/sdk/build/**
+*.tgz
```
[the remark this answers is not in the document]
Comment (Saschl) [reply .gitignore:70]: I'll remove `/**/.rollup.cache` and `tsconfig.tsbuildinfo`.
As for `/**/package-lock.json`: I see we do not have it in the project, so I'll prevent its generation for the Framework sources as well and remove the ignore. 

`msfs-avionics-mirror/src/sdk/build/**` and `*.tgz` as we probably don't want to see built sources as a diff when building locally.

I chose to generate a package as it's compatible with Windows environments after using our "setup.sh" script. If I'd do it with a relative path like described [here](https://microsoft.github.io/msfs-avionics-mirror/docs/getting-started/setting-up-your-environment) we're getting issues as npm uses a symlink (and looks like a symlink generated in the docker container isn't exactly what windows likes)

File: .github/workflows/pr.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (aguther) [reply .github/workflows/pr.yml:18]: Is that considered solved?

File: .github/workflows/pr.yml [hunk 6]
```diff
@@ -37,7 +37,7 @@ jobs:
       - name: Checkout source
         uses: actions/checkout@v2
       - name: npm install
-        run: npm install --no-optional
+        run: ./scripts/dev-env/run.sh ./scripts/setup.sh
```
[the remark this answers is not in the document]
Comment (aguther) [reply .github/workflows/pr.yml:40]: Is that considered solved?

File: .gitignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (aguther) [reply .gitignore:67]: Is that considered solved?

[the remark this answers is not in the document]
Comment (beheh) [reply .gitignore:67]: No, because the changes that Saschl mentioned he will make are still outstanding.

File: .github/workflows/pr.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (beheh) [reply .github/workflows/pr.yml:40]: No - I'm still waiting for a response.

File: .github/workflows/pr.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (beheh) [reply .github/workflows/pr.yml:18]: I would suggest we keep this open as a reminder until we've solved the msfs-avionics-mirror hosting question (submodule? build our own? keep it here? strip down? etc. etc.)

File: src/instruments/src/PFD/rollup.config.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+export default {
+    input: join(__dirname, 'instrument.tsx'),
+    output: {
+        dir: '../../../../flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/A32NX/PFD',
+        format: 'es',
+    },
+    plugins: [scss(
+        { output: '../../../../flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/A32NX/PFD/pfd.css' },
+    ),
+    // typescript({ include: ['**/*.ts', '**/*.tsx', '../shared/src/*.ts', '../fmgc/src/**/ControlLaw.ts', '../failures/src/**/*.ts'] }), resolve()],
```
Comment (beheh) [inline src/instruments/src/PFD/rollup.config.js:18]: Is this needed or not? Let's not commit comments with stale code.

File: src/instruments/src/PFD/shared/pixels.scss [hunk 8]
```diff
@@ -0,0 +1,10 @@
+.BacklightBleed {
+  content: '';
```
Comment (beheh) [inline src/instruments/src/PFD/shared/pixels.scss:2]: I don't think `content: ''` makes sense on a non-pseudo element (no :before or :after).

File: src/instruments/src/PFD/shared/utils.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
+/**
```
Comment (beheh) [inline src/instruments/src/PFD/shared/utils.tsx:1]: I think this file and PFDUtils.tsx should be merged.

File: src/instruments/src/PFD/PFD.tsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
+import { A320Failure, FailuresConsumer } from '@flybywiresim/failures';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (beheh) [inline src/instruments/src/PFD/PFD.tsx:2]: Why is this needed?

File: src/instruments/src/PFD/LandingSystemIndicator.tsx [hunk 11]
```diff
@@ -1,173 +1,309 @@
...
+                this.lsGroupRef.instance.style.display = this.lsButtonPressedVisibility ? 'inline' : 'none';
+                this.gsReferenceLine.instance.style.display = this.lsButtonPressedVisibility ? 'inline' : 'none';
+            }
+        });
+
+        /*   sub.on('rollAr').handle((r) => {
+            if (this.lsButtonPressedVisibility && !r.isNormalOperation()) {
+                this.gsReferenceLine.instance.style.display = 'inline';
+            }
+        }); */
```
Comment (beheh) [inline src/instruments/src/PFD/LandingSystemIndicator.tsx:33]: Why is this commented out?

File: src/instruments/src/PFD/FMA.tsx [hunk 12]
```diff
@@ -1,554 +1,1021 @@
...
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        this.props.isAttExcessive.sub((a) => {
+            if (a) {
+                this.cellsToHide.instance.style.visibility = 'hidden';
+                this.b1Cell.instance.displayModeChangedPath(true);
+                this.c1Cell.instance.displayModeChangedPath(true);
+                // this.D1D2Cell.instance.displayModeChangedPath(true);
```
Comment (beheh) [inline src/instruments/src/PFD/FMA.tsx:177]: Commented code.

File: src/instruments/src/PFD/FMA.tsx [hunk 13]
```diff
@@ -1,554 +1,1021 @@
...
+                this.BC1Cell.instance.displayModeChangedPath(true);
+            } else {
+                this.cellsToHide.instance.style.visibility = 'visible';
+                this.b1Cell.instance.displayModeChangedPath();
+                this.c1Cell.instance.displayModeChangedPath();
+                // this.D1D2Cell.instance.displayModeChangedPath();
```
Comment (beheh) [inline src/instruments/src/PFD/FMA.tsx:183]: Commented code.

File: src/instruments/src/PFD/FMA.tsx [hunk 14]
```diff
@@ -1,554 +1,1021 @@
...
+
+    onAfterRender(node: VNode): void {
+        super.onAfterRender(node);
+
+        const sub = this.props.bus.getSubscriber<PFDSimvars>();
+
+        sub.on('AThrMode').whenChanged().handle((athrMode) => {
+            this.athrModeSub.set(athrMode);
+
+            let text: string = '';
```
Comment (beheh) [inline src/instruments/src/PFD/FMA.tsx:285]: I don't quite understand this part and would like to learn more about when this is appropriate. Why is this specific case better solved via strings? I don't feel like I see this pattern in a lot of other places in this PR.

File: src/instruments/src/PFD/FMA.tsx [hunk 15]
```diff
@@ -1,554 +1,1021 @@
...
+            break;
+        case 21:
+            text = 'ALT CST*';
+            break;
+        case 20:
+            text = 'ALT CST';
+            break;
+        // case 18:
+        //     text = 'ALT CRZ';
+        //     break;
```
Comment (beheh) [inline src/instruments/src/PFD/FMA.tsx:594]: Why is this commented? I believe we support this mode.

File: src/instruments/src/PFD/FMA.tsx [hunk 16]
```diff
@@ -1,554 +1,1021 @@
...
+    private getText(): boolean {
+        let text: string;
+        let additionalText: string = '';
+
+        this.isShown = true;
+        switch (this.activeVerticalModeSub.get()) {
+        case 31:
+            text = 'G/S';
+            break;
+        // case 2:
```
Comment (beheh) [inline src/instruments/src/PFD/FMA.tsx:537]: Same for this and others - while we might not normally support these yet, shouldn't we just enable them so they start working? I don't quite understand why they're commented.

File: src/instruments/src/PFD/FMA.tsx [hunk 17]
```diff
@@ -1,554 +1,1021 @@
...
+            } else if (altArmed) {
+                text1 = 'ALT';
+            } else {
+                text1 = '';
+            }
+
+            let text2;
+            // case 1:
+            //     text2 = 'F-G/S';
+            //     break;
```
Comment (beheh) [inline src/instruments/src/PFD/FMA.tsx:748]: Same not as above applies here - we might as well enable this?

File: src/instruments/src/PFD/FMA.tsx (hunk 15 above)
Comment (Saschl) [reply src/instruments/src/PFD/FMA.tsx:594]: This was copied straight from the existing FMA code, but yes we can activate it. Shouldn't cause any harm.

File: src/instruments/src/PFD/FMA.tsx (hunk 16 above)
Comment (Saschl) [reply src/instruments/src/PFD/FMA.tsx:537]: This was copied straight from the existing FMA code, but yes we can activate it. Shouldn't cause any harm.

File: src/instruments/src/PFD/FMA.tsx (hunk 17 above)
Comment (Saschl) [reply src/instruments/src/PFD/FMA.tsx:748]: This one is a bit special, it was also copied but there seems no switch-case statement where it could belong to

File: src/instruments/src/PFD/FMA.tsx (hunk 15 above)
Comment (lukecologne) [reply src/instruments/src/PFD/FMA.tsx:594]: This is implemented a few lines up, within the "normal" ALT mode, by checking the lvar A32NX_FMA_CRUISE_ALT_MODE

File: src/instruments/src/PFD/PFD.tsx (hunk 10 above)
Comment (Saschl) [reply src/instruments/src/PFD/PFD.tsx:2]: It's the `FSComponent` import that is reproted as unused but needed. It's more or less the equivalent to `import 'react'` (https://microsoft.github.io/msfs-avionics-mirror/docs/getting-started/creating-your-first-component/#framework-imports). Maybe I need to adapt the eslint config so it properly recognizes it.

File: src/instruments/src/PFD/FMA.tsx (hunk 14 above)
Comment (Saschl) [reply src/instruments/src/PFD/FMA.tsx:283]: This is probably something we have to recheck in the future. Most other places in the PFD have a "static" structure where only the content or animations change. In the case of some Cells in the FMA the structure is very different dependent on the message so I was going for a "dumb" `.innerHtml` replace to properly render the content. 
I'd like to revisit this in the future. For example I could add all messages (or all that share a common element structure) and update/hide them accordingly. But that needs some trial and error on my side

File: src/instruments/src/PFD/FMA.tsx (hunk 16 above)
Comment (beheh) [reply src/instruments/src/PFD/FMA.tsx]: Ah, in that case I don't mind then, we can enable it whenever we need it - but we might as well do now :+1:

File: .gitignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (beheh) [reply .gitignore:70]: The package-lock.json part has been solved with now with #6898.

File: msfs-avionics-mirror/src/sdk/.npmrc [hunk 18]
```diff
@@ -0,0 +1 @@
+package-lock=false
```
Comment (beheh) [inline msfs-avionics-mirror/src/sdk/.npmrc:1]: As we've switched the main repository, how about we now also generate a package-lock.json here?

File: src/instruments/src/PFD/shared/PFDSimvarPublisher.tsx [hunk 19]
```diff
@@ -0,0 +1,358 @@
+import { EventBus, SimVarDefinition, SimVarValueType, SimVarPublisher } from 'msfssdk';
+
+export interface PFDSimvars {
```
Comment (beheh) [inline src/instruments/src/PFD/shared/PFDSimvarPublisher.tsx:3]: I assume these should match PFDVars 1:1? Some of the casing is slightly different.

There's also a way to enforce this with Typescript, but I'm struggling to write the precise type right now (you'd need a separate type with the boolean keys or something).

**Review by beheh — CHANGES_REQUESTED** [review]
I gave this another review, and overall I think we're nearly there. As per our chat on Discord, it seems at this stage it is appropriate to proceed with the `msfs-avionics-mirror` code in the repository, as it seems WT has no interest in maintaining a package.

File: package.json [hunk 20]
```diff
@@ -98,6 +100,7 @@
     "classnames": "^2.2.6",
     "immer": "^9.0.3",
     "lodash": "^4.17.20",
+    "msfssdk": "file:msfs-avionics-mirror/src/sdk/build/msfssdk-0.1.0.tgz",
```
[the remark this answers is not in the document]
Comment (Saschl) [reply package.json:103]: Is very much possible (it's recommended like this in the readme even). Buuut the way npm handles this (via symlinks) doesn't really work when running `setup.sh`. I assume it's due to the difference between the Unix and Windows world.

TLDR;
When running `setup.sh` the IDE will not find the Framework and complain. When running npm install locally, the IDE is happy but the `build.sh` won't work.

File: src/instruments/src/PFD/shared/PFDSimvarPublisher.tsx (hunk 19 above)
Comment (Saschl) [reply src/instruments/src/PFD/shared/PFDSimvarPublisher.tsx:3]: I'll have a look (and at the very least unify the casing) 👍

File: src/instruments/src/PFD/SpeedIndicator.tsx [hunk 21]
```diff
@@ -1,351 +1,919 @@
...
+            this.upperBoundRef.instance.style.visibility = 'visible';
+            this.lowerBoundRef.instance.style.visibility = 'hidden';
+            this.speedTargetRef.instance.style.visibility = 'hidden';
+            this.currentVisible = this.upperBoundRef;
+        } else if (this.speedState.speed.value - currentTargetSpeed < -DisplayRange && !this.decelActive) {
+            this.lowerBoundRef.instance.style.visibility = 'visible';
+            this.upperBoundRef.instance.style.visibility = 'hidden';
+            this.speedTargetRef.instance.style.visibility = 'hidden';
+            this.currentVisible = this.lowerBoundRef;
+        } else if (!this.decelActive) {
```
Comment (lukecologne) [inline src/instruments/src/PFD/SpeedIndicator.tsx:817]: The selected speed bug should not be always hidden when DECEL is active, instead, it should only show when it is
in range.
```suggestion
        } else if (Math.abs(airspeed - targetSpeed) < DisplayRange) {
```

File: src/instruments/src/PFD/SpeedIndicator.tsx [hunk 22]
```diff
@@ -1,351 +1,919 @@
...
+            this.speedState.isMach = s;
+            this.needsUpdate = true;
+        });
+
+        sub.on('targetSpeedManaged').whenChanged().handle((s) => {
+            this.speedState.managedTargetSpeed = s;
+            this.needsUpdate = true;
+        });
+
+        sub.on('autoBrakeActive').whenChanged().handle((a) => {
```
Comment (lukecologne) [inline src/instruments/src/PFD/SpeedIndicator.tsx:762]:
```suggestion
        sub.on('autoBrakeDecel').whenChanged().handle((a) => {
```

File: src/instruments/src/PFD/FMA.tsx [hunk 23]
```diff
@@ -1,567 +1,1178 @@
...
-        className = 'White BlinkInfinite';
-    } else if (engineMessage === 5) {
-        text = 'LVR ASYM';
-        className = 'Amber';
-    } else if (ABrkMode === 3 && !ABrkActive && !AB3Message) {
-        text = 'BRK MAX';
-        className = 'Cyan';
-    } else {
-        return null;
+    private setAutoBrakeText(newAutobrakeActive: boolean) {
```
Comment (lukecologne) [inline src/instruments/src/PFD/FMA.tsx:444]: The A/BRK and A/THR text setting should not be seperate like that.
If I'm not mistaken, it is theoretically possible now that an A/THR mode is still active, while the A/BRK active mode is shown. (Yes, this is a very remote possibility, but nevertheless it should be considered)

File: src/instruments/src/PFD/FMA.tsx [hunk 24]
```diff
@@ -1,567 +1,1178 @@
...
+
+        sub.on('autoBrakeActive').whenChanged().handle((am) => {
+            if (am) {
+                this.autoBrkRef.instance.style.visibility = 'hidden';
+            } else {
+                this.autoBrkRef.instance.style.visibility = 'visible';
+            }
+        });
+
+        sub.on('AThrMode').whenChanged().handle((athrMode) => {
```
Comment (lukecologne) [inline src/instruments/src/PFD/FMA.tsx:277]: Same as comment above.

File: src/instruments/src/PFD/FMA.tsx [hunk 25]
```diff
@@ -1,567 +1,1164 @@
...
+        sub.on('autoBrakeMode').whenChanged().handle((am) => {
+            switch (am) {
+            case 0:
+                this.text.set('');
+                break;
+            case 1:
+                this.text.set('BRK LO');
+                break;
+            case 2:
+                this.text.set('BRK MED');
```
Comment (lukecologne) [inline src/instruments/src/PFD/FMA.tsx:258]: There are spaces missing here at the end of the message, which are there to move the whole message to the left.
I think it was only one space at the end, but better look at the old message string to confirm

File: src/instruments/src/PFD/shared/PFDSimvarPublisher.tsx (hunk 19 above)
Comment (Saschl) [reply src/instruments/src/PFD/shared/PFDSimvarPublisher.tsx:3]: Casing should be the same now, unfortunately couldn't find the type to enforce it

**Comment by Saschl** [comment]
I think we can give this another review 👍

File: package.json [hunk 26]
```diff
@@ -113,11 +116,12 @@
     "react-use-websocket": "^2.9.1",
     "redux": "^4.0.5",
     "redux-thunk": "^2.3.0",
-    "rollup-plugin-scss": "^2.6.1",
-    "sass": "~1.30.0",
+    "rollup-plugin-import-css": "^3.0.2",
```
Comment (beheh) [inline package.json:119]: Is this actually used? I can't find it anywhere, specifically not in the rollup config file.

**Review by beheh — APPROVED** [review]
Two more nitpicks, but generally we can proceed here. The scope is massive of course, and it would be nice if we could eventually get this code test covered, but that applies to most of our JS code today.

Great job Saschl!

File: package.json (hunk 26 above)
Comment (Saschl) [reply package.json]: Removed 👍

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #6543
Tier of Testing  : 2
Date             : 06/04/2022

Testing Process: 
A bunch of flights on exp version.

The behaviour of the PFD should be the same as it is with the development version currently. - **OK**
Make sure there is no missing symbols, texts (especially the FMA). **-OK**
You might not always see higher FPS, but the experience should be smoother in comparison to the development version. **OK**, similar fps but way less stutters.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Comment by Saschl** [comment]
Adding to the report, this PR has been available on the experimental version for over a month already with no recent reports about issues.

**Comment by alexr4339** [comment]
Quality Assurance Trainee/Tester Report

Discord : Alex R#6663
Object of testing: #6543
Tier of Testing : 1
Date : 07/04/2022

Testing Process:



Spawned on the runway and did not insert anything (no departure or arrival). As I took off I noticed that it did not show the Nav and Climb modes roughly for a minute. Roughly a minute after liftoff those modes were shown on the PFD and from this point on it worked exactly as expected. I'd assume it took some time to properly initialise, something that would go unnoticed under normal C&D flights

Negatives:
Not showing Nav/Climb modes for some time after starting your flight

Testing Results:
Not Passed

Media:

**Comment by Saschl** [comment]
> Quality Assurance Trainee/Tester Report
> 
> Discord : Alex R#6663 Object of testing: #6543 Tier of Testing : 1 Date : 07/04/2022
> 
> Testing Process:
> 
> Spawned on the runway and did not insert anything (no departure or arrival). As I took off I noticed that it did not show the Nav and Climb modes roughly for a minute. Roughly a minute after liftoff those modes were shown on the PFD and from this point on it worked exactly as expected. I'd assume it took some time to properly initialise, something that would go unnoticed under normal C&D flights
> 
> Negatives: Not showing Nav/Climb modes for some time after starting your flight
> 
> Testing Results: Not Passed
> 
> Media:  

At the attitude seen in the first picture, the modes are not supposed to be shown, but I saw one thing, the white border should not appear after all then. This is fixed.

**Comment by aguther** [comment]
I've checked the reported issue from the QA tester and it works now.
